src/pages/Exchange/steps/base.tsx

```diff
@@ -100,25 +101,32 @@ const getBalance = async (
   const swapFeeToken = ((swapFeeUsd / Number(token.price)) * 0.9).toFixed(`${toInteger(token.price)}`.length);
 
   const src_coin = exchange.transaction.tokenSelected.src_coin;
-
   if (exchange.token === TOKEN.ERC20) {
     if (!userMetamask.erc20TokenDetails) {
+
       eth.maxAmount = '0';
       eth.minAmount = '0';
     }
     scrt.maxAmount = user.balanceToken[src_coin] ? user.balanceToken[src_coin] : '0';
     scrt.minAmount = `${Math.max(Number(swapFeeToken), Number(token.display_props.min_from_scrt))}` || '0';
-    eth.maxAmount = userMetamask.erc20Balance;
-    eth.minAmount = userMetamask.erc20BalanceMin || '0';
+    eth.maxAmount = exchange.network === NETWORKS.BSC ? userMetamask.balanceToken[src_coin] : userMetamask.erc20Balance || '0';
+    eth.minAmount = exchange.network === NETWORKS.BSC ? userMetamask.balanceTokenMin[src_coin] : userMetamask.erc20BalanceMin || '0';
   } else {
     scrt.maxAmount =
       !user.balanceToken[userMetamask.getNetworkFullName()] ||
-      user.balanceToken[userMetamask.getNetworkFullName()].includes(unlockToken)
+        user.balanceToken[userMetamask.getNetworkFullName()].includes(unlockToken)
         ? '0'
         : user.balanceToken[userMetamask.getNetworkFullName()];
     scrt.minAmount = `${Math.max(Number(swapFeeToken), Number(token.display_props.min_from_scrt))}` || '0';
+
+    if (exchange.network === NETWORKS.BSC) {
+      scrt.maxAmount = user.balanceToken[src_coin] ? user.balanceToken[src_coin] : '0';
+    }
```

Comment:
Why is this BSC specific? We will want to add more networks here as well. Try to make this generic to any chain and not add specific BSC logic.

Also having this code here makes me think that it would be better to modify the above logic setting scrt.maxAmount, and not add this as a patch

Reply:
resolved on last commit